xla/hlo/analysis/hlo_replication_analysis_test.cc

```diff
@@ -673,6 +673,81 @@ ENTRY entry {
       FindInstruction(module.get(), "dynamic-slice"), {}, replica_groups_2));
 }
 
+TEST_F(HloReplicationAnalysisTest, PartiallyReplicatedAllGatherFlattenedID) {
+  const std::string module_str = R"(
+HloModule PartiallyReplicatedAllGatherFlattenedID
+
+ENTRY entry {
+  param = s32[2] parameter(0)
+  all-gather0 = s32[4] all-gather(param), dimensions={0}, replica_groups={{0,2},{4,6},{1,3},{5,7}}, channel_id=1, use_global_device_ids=true
```

Comment:
Remembering what these all-gather replica groups is hard as I scroll down to the rest of the test. I kept having to scroll back up again.

Instead, group all the relevant parts of each instruction being tested. Perhaps by having a lambda taking in the `replica_groups` of the instruction, the `replica_groups` pass to `HloInstructionIsReplicatedAt`, and the value of cross_partition_spmd, and returning whether HloInstructionIsReplicatedAt returned true or not. Then all the information needed to determine whether it should be replicated or not is only a single line.